stock_barcodes/static/src/kanban_renderer.esm.js

```diff
@@ -0,0 +1,163 @@
+
+import {KanbanRenderer} from "@web/views/kanban/kanban_renderer";
+import {isAllowedBarcodeModel} from "./barcodes_models_utils.esm";
+import {patch} from "@web/core/utils/patch";
+import {useBus} from "@web/core/utils/hooks";
+import {useHotkey} from "@web/core/hotkeys/hotkey_hook";
+import {useRef} from "@odoo/owl";
+
+patch(KanbanRenderer.prototype, "add hotkey", {
+    setup() {
+        const rootRef = useRef("root");
+        useHotkey(
+            "Enter",
+            ({target}) => {
+                if (!target.classList.contains("o_kanban_record")) {
+                    return;
+                }
+
+                // Open first link
+                let firstLink = null;
+                if (isAllowedBarcodeModel(this.props.list.resModel)) {
+                    firstLink = target.querySelector(
+                        ".oe_kanban_action_button,.oe_btn_quick_action"
+                    );
+                }
+                if (!firstLink) {
+                    firstLink = target.querySelector(
+                        ".oe_kanban_global_click, a, button"
+                    );
+                }
+                if (firstLink && firstLink instanceof HTMLElement) {
+                    firstLink.click();
+                }
+                return;
+            },
+            {area: () => rootRef.el}
+        );
+
+        this._super(...arguments);
+
+        if (isAllowedBarcodeModel(this.props.list.resModel)) {
+            if (this.env.searchModel) {
+                useBus(this.env.searchModel, "focus-view", () => {
+                    const {model} = this.props.list;
+                    if (model.useSampleModel || !model.hasData()) {
+                        return;
+                    }
+                    const cards = Array.from(
+                        rootRef.el.querySelectorAll(".o_kanban_record")
+                    );
+                    const firstCard = cards.find(
+                        (card) =>
+                            card.querySelectorAll("button[name='action_barcode_scan']")
+                                .length > 0
+                    );
+                    if (firstCard) {
+                        // Focus first kanban card
+                        firstCard.focus();
+                    }
+                });
+            }
+        }
+    },
+    // eslint-disable-next-line complexity
+    focusNextCard(area, direction) {
+        const {isGrouped} = this.props.list;
+        const closestCard = document.activeElement.closest(".o_kanban_record");
+        if (!closestCard) {
+            return;
+        }
+        const groups = isGrouped
+            ? [...area.querySelectorAll(".o_kanban_group")]
+            : [area];
+        let cards = [...groups]
+            .map((group) => [...group.querySelectorAll(".o_kanban_record")])
+            .filter((group) => group.length);
+
+        if (isAllowedBarcodeModel(this.props.list.resModel)) {
```

Comment:
The code is not returning in the `if` statement it's recomputing the `cards` variable.
I could call super and return its value but I would still have duplicate that code because nothing is returned from the parent function. I still need to compute the candidate `cards` still filter them and focus the correct one.

If you have a way to do it better I would need more information because I'm not sure how to go about it.